go/state/state_db.go

```diff
@@ -534,6 +541,14 @@ func (s *stateDB) CreateAccount(addr common.Address) {
 	})
 }
 
+func (s *stateDB) CreateContract(addr common.Address) {
+	s.createAccountIfNotExists(addr)
+	s.createdContracts[addr] = true
```

Comment:
Although there are external mechanisms that should prevent this, we should be defensive in the implementation of this function. Thus, it should be checked whether the given address is already in the `createdContracts` set, and if so, the function should ignore the extra call. Otherwise, snapshot restorations may reproduce wrong values.